packages/wasm-parser/src/decoder.js

```diff
@@ -319,10 +318,26 @@ export function decode(ab: ArrayBuffer, opts: DecoderOpts): Program {
       if (type == types.func) {
         dump([type], "func");
 
-        const paramValtypes: Array<Valtype> = parseVec(b => valtypes[b]);
+        const paramValtypes: Array<Valtype> = parseVec(b => {
+          const t = valtypes[b];
```

Comment:
well this part can be a separate method I guess.

Reply:
Yeah I agree, it wrote this to have a better error message when debuging